Per-module logging

andreyaksenov opened the pull request:
- Added the description for the [box.cfg.log_modules](https://docs.d.tarantool.io/en/doc/2.11-per-module-logging/reference/configuration/#confval-log_modules) configuration property.
- Added the [log.new()](https://docs.d.tarantool.io/en/doc/2.11-per-module-logging/reference/reference_lua/log/#log-new) function description to the `log` module API reference.
- Added a note that environment variables are taken into account if [log.cfg()](https://docs.d.tarantool.io/en/doc/2.11-per-module-logging/reference/reference_lua/log/#lua-function.log.cfg) is called before `box.cfg()`.

Note that doc code snippets are included from testable examples placed in the [doc/code_snippets](https://github.com/tarantool/doc/tree/latest/doc/code_snippets) folder for making sure they are correct and runnable.

andreyaksenov committed:
Per-module logging

Gumix reviewed (commented):
The following points from #3264 are missed:

It is possible to set custom log levels for files (modules) that just use default logger without creating a new instance by `log.new()`
> The name of a module is determined automatically during the execution of require('log') in the module's source code. The name is derived from its filename, including a part of the path.

Example: https://github.com/tarantool/tarantool/blob/fa6ae4a3c366f7fec3fd57403bca313cf762904e/test/box-luatest/gh_3211_module/testmod.lua

https://github.com/tarantool/tarantool/blob/fa6ae4a3c366f7fec3fd57403bca313cf762904e/test/box-luatest/gh_3211_per_module_log_level_test.lua#L175-L179

> There is a special module name "tarantool", which allows to set log level for Tarantool core messages. To be more precise, it sets log level for all messages logged from the non-Lua code, including user C modules.

Example: https://github.com/tarantool/tarantool/blob/fa6ae4a3c366f7fec3fd57403bca313cf762904e/test/box-luatest/gh_3211_per_module_log_level_test.lua#L190-L194

andreyaksenov commented:
Thanks! Updated the docs as follows to reflect missed points:
- Updated the [box.cfg.log_modules](https://docs.d.tarantool.io/en/doc/2.11-per-module-logging/reference/configuration/#confval-log_modules) description to show that different module types are supported.
- Added 3 examples for all module types (file, custom, and tarantool).
- Added the [modules](https://docs.d.tarantool.io/en/doc/2.11-per-module-logging/reference/reference_lua/log/#log-cfg) property to the `log.cfg({})` description.

xuniq commented on doc/reference/reference_lua/log.rst:
```diff
@@ -85,6 +92,11 @@ Below is a list of all ``log`` functions.
+
+        Note that calling ``log.cfg()`` before ``box.cfg()`` takes into account
+        logging options specified using :ref:`environment variables <box-cfg-params-env>`,
+        such as ``TT_LOG``, ``TT_LOG_LEVEL``, and so on.
```
```suggestion
        such as ``TT_LOG`` and ``TT_LOG_LEVEL``.
```

xuniq commented on doc/reference/reference_lua/log.rst:
```diff
@@ -146,8 +158,46 @@ Below is a list of all ``log`` functions.
+        :lines: 9-13
+        :dedent:
+
+    To create the ``module1`` and ``module1`` modules, call the ``new()`` function:
```
```suggestion
    To create the ``module1`` and ``module2`` loggers, call the ``new()`` function:
```
